odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+"""Single-photon emission computerized tomography (SPECT) projectors """
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+# External
+try:
+    from NiftyPy.NiftyRec import (SPECT_project_parallelholes,
+                                  SPECT_backproject_parallelholes)
+    NIFTYREC_AVAILABLE = True
+except ImportError:
+    NIFTYREC_AVAILABLE = False
+
+import numpy as np
+
+# Internal
+import odl
+from odl.operator.operator import Operator
+
+
+_SUPPORTED_IMPL = ('niftyrec')
+__all__ = ('SpectProject', 'SpectBackproject')
+
+
+class SpectProject(Operator):
+    """Spect projector using backend NiftyRec"""
+    def __init__(self, dom, geometry, attenuation, psf, use_gpu=False,
+                 impl='niftyrec', **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Discretized space, the domain of the forward projector
+        geometry : `Geometry`
+            The geometry of the transform. An ODL geometry instance
+            that contains information about the operator domain
+        attenuation : `DiscreteLpVector`
```

Comment:
We should accept anything that can be *converted* into an adequate space element. The adequate space could be the same as this operator's domain or some other space with the same domain of definition.
The simplest assumption would be the former, which would then simply mean that [this line](https://github.com/odlgroup/odl/pull/391/files#diff-f72b925327f97605e512656ea6a10472R85) is replaced with 
```python
self._attenuation = self.domain.element(attenuation)
```
Support for finer or coarser spaces can be added later. Should be simple with the `Resampling` operator, however. Or even doable by hand.